File: KIARA.js
// const input = require('fs').readFileSync('/dev/stdin', 'utf8');
const input = require("fs").readFileSync("./input.txt", "utf8");
const lines = input.split("\n");
let nextLine = 0;

const numWords = lines[nextLine++];

const initialLetters = new Set();
const words = [];

const areSetsEqual = (a, b) => [...a].every((value) => b.has(value));

for (let i = 0; i < numWords; i++) {
    const word = lines[nextLine++].slice(0, -1);
    words.push(new Set([...word]));
    initialLetters.add(word[0]);
}

let exit = false;
for (let word of words) {
    if (areSetsEqual(word, initialLetters)) {
        console.log("Y");
        exit = true;
        break;
    }
}

if (!exit) console.log("N");

File: KIARA.py
# 3369

t=int(input())
l=[]
ini={}
for i in range(t):
    inp = input()
    ini[inp[0]] = 1
    a={}
    for s in list(inp):
        a[s] = 1
    l.append(a)

for palavra in l:
    if palavra == ini:
        print('Y')
        exit()

print("N")